code/datums/components/boomerang.dm

```diff
@@ -0,0 +1,76 @@
+#define BOOMERANG_REBOUND_INTERVAL (1 SECONDS)
+/**
+ * If an ojvect is given the boomerang component, it should be thrown back to the thrower after either hitting it's target, or landing on the thrown tile.
+ * Thrown objects should
+ */
+/datum/component/boomerang
+	///How far should the boomerang try to travel to return to the thrower?
+	var/boomerang_throw_range = 3
+	///"Thrownthing" datum for the most recent throw.
+	var/datum/thrownthing/thrown_boomerang
+	///If this boomerang is thrown, does it re-enable the throwers throw mode?
+	var/thrower_easy_catch_enabled = FALSE
+	///The object the component's being applied to.
+	var/obj/item/true_parent
+	COOLDOWN_DECLARE(last_boomerang_throw)
+
+/datum/component/boomerang/Initialize(_boomerang_throw_range, _thrower_easy_catch_enabled)
+	. = ..()
+	if(!isitem(parent)) //Only items support being thrown around like a boomerang, feel free to make this apply to humans later on.
+		return COMPONENT_INCOMPATIBLE
+	true_parent = parent
+
+	//Assignments
+	if(_boomerang_throw_range)
+		boomerang_throw_range = _boomerang_throw_range
+	if(_thrower_easy_catch_enabled)
+		thrower_easy_catch_enabled = _thrower_easy_catch_enabled
```

Comment:
- Don't bother saving `true_parent`, just cast it when needed
- underscore parameters are out of style, now parameters are usually just the same as the var name, and we use `src.var = var` to assign them (see the transforming component for an example)